code/modules/mob/mob.dm

```diff
@@ -510,6 +511,40 @@
 
 	return TRUE
 
+/**
+ * handle_eye_contact() is called when we examine() something. If we examine an alive mob with a mind who has examined us in the last 2 seconds within 5 tiles, we make eye contact!
+ *
+ * Note that if either party has their face obscured, the other won't get the notice about the eye contact
+ * Also note that examine_more() doesn't proc this or extend the timer, just because it's simpler this way and doesn't lose much.
+ * The nice part about relying on examining is that we don't bother checking visibility, because we already know they were both visible to each other within the last second, and the one who triggers it is currently seeing them
+ */
+/mob/proc/handle_eye_contact(mob/living/examined_mob)
+	return
+
+/mob/living/handle_eye_contact(mob/living/examined_mob)
+	if(!istype(examined_mob) || src == examined_mob || examined_mob.stat >= UNCONSCIOUS || !client)
+		return
+
+	var/imagined_eye_contact = FALSE
+	if(!LAZYACCESS(examined_mob.client?.recent_examines, src))
+		// even if you haven't looked at them recently, if you have the shift eyes trait, they may still imagine the eye contact
+		if(HAS_TRAIT(examined_mob, TRAIT_SHIFTY_EYES) && prob(10 - get_dist(src, examined_mob)))
+			imagined_eye_contact = TRUE
+		else
+			return
+
+	if(get_dist(src, examined_mob) > EYE_CONTACT_RANGE)
+		return
+
+	// check to see if their face is blocked or, if not, a signal blocks it
+	if(examined_mob.is_face_visible() && SEND_SIGNAL(src, COMSIG_MOB_EYECONTACT, examined_mob, TRUE) != COMSIG_BLOCK_EYECONTACT)
+		var/msg = span_smallnotice("You make eye contact with [examined_mob].")
+		addtimer(CALLBACK(GLOBAL_PROC, .proc/to_chat, src, msg), 3) // so the examine signal has time to fire and this will print after
```

Comment:
Instead of all these callbacks to wait for the examine text, couldn't you just handle eye contact *after* the examine text is scheduled in `/mob/verb/examinate()`?

Reply:
truthfully, this code was mostly just copy pasted from tg without much consideration to the hows and whys outside of changing how some functions worked to match our code. to my understanding it's so that handle eye contact can handle any weird cases like with shifty eyes giving a perceived response, or later on adding more unique interactions with eye contact where it may or may not work, or altering the message later on. I'm not sure if that totally answers your question so i'm sorry if i missed the point

Reply:
Oh don't worry about it, I just wanted to mark that I noticed this could be worked around if refactored. I don't think it's terribly important, just a very tiny performance loss at worst.